src/platform/react-hooks/src/ChannelProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useLayoutEffect, useEffect } from 'react';
+import * as Ably from 'ably';
+import { getContext } from './AblyProvider.js';
+import * as Utils from './utils/utils.js';
+import { channelOptionsWithAgent } from './AblyReactHooks.js';
+
+interface ChannelProviderProps {
+  id?: string;
+  channelName: string;
+  options?: Ably.Types.ChannelOptions;
+  deriveOptions?: Ably.Types.DeriveOptions;
+  children?: React.ReactNode | React.ReactNode[] | null;
+}
+
+export const ChannelProvider = ({
+  id = 'default',
+  channelName,
+  options,
+  deriveOptions,
+  children,
+}: ChannelProviderProps) => {
+  const { client, _channelToOptions } = React.useContext(getContext(id));
+
+  const channel = deriveOptions
+    ? client.channels.getDerived(channelName, deriveOptions)
```

Comment:
I don't really like this, because derived channel has a different name. And I don't know what are the use-cases, does it make sense to have regular channel and derived at the same time?

Reply:
refactored a bit, now you need to setup deriveOptions on provider level only, and then use in hook just by name